src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs

```diff
@@ -92,4 +92,26 @@ public override bool IsInvalid
             get { return handle == IntPtr.Zero; }
         }
     }
+
+    [SecurityCritical]
+    internal sealed class SafeSharedX509StackHandle : SafeHandle
+    {
+        internal static readonly SafeSharedX509StackHandle InvalidHandle = new SafeSharedX509StackHandle();
+
+        private SafeSharedX509StackHandle() :
+            base(IntPtr.Zero, ownsHandle: false)
+        {
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            SetHandle(IntPtr.Zero);
```

Comment:
Thanks for the explanation.

> I'm wondering if this SafeHandle needs a field reference to the owning SafeHandle (to ensure it's lifetime WRT finalization), have IsInvalid take that handle's IsInvalid into account, and ReleaseHandle just set that field to null.

I'm actually not sure what the best practice is for SafeHandles that wrap portions of other SafeHandles.  @jkotas?  Maybe the creation of the "child" handle AddRef's the parent and its ReleaseHandle ReleaseRef's the parent?